README.md

```diff
@@ -2,5 +2,435 @@
 ==========
 [![Build Status](https://github.com/razz-team/eva/actions/workflows/main_build.yml/badge.svg)](https://github.com/razz-team/eva/actions)
 [<img src="https://img.shields.io/maven-central/v/team.razz.eva/eva-domain.svg?label=latest%20release"/>](https://search.maven.org/search?q=g:team.razz.eva%20OR%20g:team.razz.eva)
+
+Welcome to Eva! It is a Kotlin open-source framework, which helps you to write your code in DDD style and using CQRS approach.
+
+
+## Getting started
+
+Declare Eva dependencies in your project:
+
+```kotlin
+dependencies {
+    implementation("team.razz.eva:eva-uow:$eva_version")
+    implementation("team.razz.eva:eva-repository:$eva_version")
+    implementation("team.razz.eva:eva-persistence-jdbc:$eva_version")
+}
+```
+
+### Model
+Define events for your business domain.
+You can read more about domain and integration events [here](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/domain-events-design-implementation). 
+```kotlin
+sealed class WalletEvent : ModelEvent<Wallet.Id> {
+
+    override val modelName = "Wallet"
+
+    data class Created(
+        override val modelId: Wallet.Id,
+        val currency: Currency,
+        val amount: ULong,
+        val expireAt: Instant
+    ) : WalletEvent(), ModelCreatedEvent<Wallet.Id> {
+        override fun integrationEvent() = buildJsonObject {
+            put("currency", currency.currencyCode)
+            put("amount", amount.toLong())
+            put("expireAt", expireAt.epochSecond)
+        }
+    }
+
+    data class Deposit(
+        override val modelId: Wallet.Id,
+        val walletAmount: ULong,
+        val depositAmount: ULong
+    ) : WalletEvent(), ModelCreatedEvent<Wallet.Id> {
+        override fun integrationEvent() = buildJsonObject {
+            put("walletAmount", walletAmount.toLong())
+            put("depositAmount", depositAmount.toLong())
+        }
+    }
+}
+```
+
+Define a model and methods that change model's state.
+On any model's modification we should raise an event about it.
+```kotlin
+class Wallet(
+    id: Id,
+    val currency: Currency,
+    val amount: ULong,
+    val expireAt: Instant,
+    entityState: EntityState<Id, WalletEvent>
+) : Model<Wallet.Id, WalletEvent>(id, entityState) {
+
+    data class Id(override val id: UUID) : ModelId<UUID>
+
+    fun deposit(toDeposit: ULong) = Wallet(
+        amount = amount - toDeposit,
+        currency = currency,
+        id = id(),
+        expireAt = expireAt,
+        entityState = entityState()
+            .raiseEvent(WalletEvent.Deposit(id(), amount, toDeposit))
+    )
+}
+```
+
+### Unit of work
+We need *queries* interface, so we can query our existing models
```

Comment:
We don't *need* it, we are perfectly fine without such abstraction.

Reply:
agree. just wanted to show how to solve some real case and interact with queries. will change this sentence.